flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunner.java

```diff
@@ -112,6 +114,13 @@ public TaskManagerRunner(
 		// Initialize the TM metrics
 		TaskExecutorMetricsInitializer.instantiateStatusMetrics(taskManagerMetricGroup, taskManagerServices.getNetworkEnvironment());
 
+		HeartbeatManagerImpl heartbeatManager = new HeartbeatManagerImpl(
+				taskManagerConfiguration.getTimeout().toMilliseconds(),
+				resourceID,
+				executor,
+				Executors.newSingleThreadScheduledExecutor(),
```

Comment:
Yes, I totally agree with that and the advantages can avoid bringing extra thread pool and run all the PRC messages in the uniform executor if **RPCService** can handle well.